Sources/AsyncHTTPClient/ConnectionPool/HTTPConnectionPool+Factory.swift

```diff
@@ -243,7 +303,11 @@ extension HTTPConnectionPool.ConnectionFactory {
         case .https:
             var tlsConfig = self.tlsConfiguration
             // since we can support h2, we need to advertise this in alpn
-            tlsConfig.applicationProtocols = ["http/1.1" /* , "h2" */ ]
+            if self.allowHTTP2Connections {
+                tlsConfig.applicationProtocols = ["http/1.1", "h2"]
```

Comment:
The protocols offered in ALPN are in descending preference order. Should we offer h2 first and prefer it over http/1.1? (https://datatracker.ietf.org/doc/html/rfc7301#section-3.1)